ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/policy-maps-qos.j2

```diff
@@ -2,16 +2,14 @@
 {% for policy_map in policy_maps.qos | arista.avd.natural_sort %}
 !
 policy-map type quality-of-service {{ policy_map }}
-{%     for class in policy_maps.qos[policy_map].classes | arista.avd.natural_sort %}
+{%     for class in policy_maps.qos[policy_map].classes | arista.avd.default([]) %}
    class {{ class }}
-{%         if policy_maps.qos[policy_map].classes[class].set is arista.avd.defined %}
-{%             for set in policy_maps.qos[policy_map].classes[class].set %}
-{%                 set cli_set = set | replace('_','-') | lower %}
-{%                 if cli_set in ['cos', 'dscp', 'traffic-class', 'drop-precedence']  %}
+{%         for set in policy_maps.qos[policy_map].classes[class].set | arista.avd.default([]) %}
+{%             set cli_set = set | replace('_','-') | lower %}
+{%             if cli_set in ['cos', 'dscp', 'traffic-class', 'drop-precedence'] %}
       set {{ cli_set }} {{ policy_maps.qos[policy_map].classes[class].set[set] }}
-{%                 endif %}
-{%             endfor %}
+{%             endif %}
+{%         endfor %}
    !
```

Comment:
Need to check this delimiter as it was inside ```if policy_maps.qos[policy_map].classes[class].set is arista.avd.defined``` previously.

Reply:
good idea. Will produce config consistent with EOS

Reply:
fixed!